cinder/volume/drivers/vmware/fcd.py

```diff
@@ -147,7 +149,14 @@ def delete_volume(self, volume):
             LOG.warning("FCD provider location is empty for volume %s",
                         volume.id)
         else:
-            self._delete_fcd(volume.provider_location)
+            try:
+                self._delete_fcd(volume.provider_location)
+            except vexc.VimException as ex:
+                if "could not be found" in str(ex):
+                    LOG.warning("FCD deletion failed for %(vol)s not found. "
+                                "delete_volume is considered successful.")
```

Comment:
Missing the `vol` argument to fill it in the string.

Reply:
fixed